src/lib/consensus/proof_of_stake.ml

```diff
@@ -66,6 +66,24 @@ module Constants = struct
       Time.Span.of_ms Int64.Infix.(Slot.duration_ms * int64_of_uint32 size)
   end
 
+  module Checkpoint_window = struct
+    let per_year = 12
+
+    let slots_per_year =
+      let one_year_ms =
+        Core.Time.Span.(to_ms (of_day 365.)) |> Float.to_int |> Int.to_int64
+      in
+      Int64.Infix.(one_year_ms / Slot.duration_ms) |> Int64.to_int
```

Comment:
what would a more explicit assertion look like to you?

Reply:
A more explicit assertion here would be a toplevel assertion about the configuration values provided during compile time. Something that ensures that the slot value provided is valid and works with this math that's a little higher up and easier to grok on it's own since it's explicit.